### PR TITLE
use domain.Offset instead of String in ContractsFetch functions 

### LINKED ISSUE #11159: use domain.Offset instead of String in ContractsFetch functions
There's a bunch of back-and-forth between `domain.Offset` and `String` in [AcsTxStreams](https://github.com/digital-asset/daml/blob/9fd8182bbb6c7ee85b65a7edbeba68ca5ac9b901/ledger-service/fetch-contracts/src/main/scala/fetchcontracts/AcsTxStreams.scala) and [ContractsFetch](https://github.com/digital-asset/daml/blob/9fd8182bbb6c7ee85b65a7edbeba68ca5ac9b901/ledger-service/http-json/src/main/scala/com/digitalasset/http/ContractsFetch.scala). This was originally meant to keep the general-purpose utilities independent of the json-api domain model so that they could be easily reused, but `domain.Offset` got pulled into the json-api-independent `fetch-contracts` library in #11057 anyway, so we can abandon the pretense and simplify the glue.

### PR DESCRIPTION
**Pull request by skisel**
Potentially fixes #11159

Not sure if I did it fully as expected in the issue :)

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by stefanobaghino-da** [comment]
@S11001001 can you review this, please?

File: ledger-service/fetch-contracts/src/main/scala/fetchcontracts/AcsTxStreams.scala [hunk 1]
```diff
@@ -93,29 +94,26 @@ private[daml] object AcsTxStreams {
...
       import domain.Offset.`Offset ordering`
       val lastTxOff = b add last(LedgerBegin: Off)
-      type EndoBookmarkFlow[A] = Flow[BeginBookmark[A], BeginBookmark[A], NotUsed]
-      val maxOff = b add domain.Offset.tag.unsubst[EndoBookmarkFlow, String](
-        max(LedgerBegin: BeginBookmark[domain.Offset])
-      )
+      val maxOff = b add max(LedgerBegin: Off)
       // format: off
       discard { txnSplit.in <~ txns <~ dupOff }
       discard {                        dupOff                                       ~> mergeOff ~> maxOff }
```
Comment (S11001001) [inline ledger-service/fetch-contracts/src/main/scala/fetchcontracts/AcsTxStreams.scala:115]: Whitespace fixing for the change on the next line. (Hard to see the difference in this comment; just align the `mergeOff`s.)

```suggestion
      discard {                        dupOff                                ~> mergeOff ~> maxOff }
```

**Review by S11001001 — APPROVED** [review]
Just a tiny whitespace nitpick, everything else looks perfect. Thanks!

**Comment by stefanobaghino-da** [comment]
> Just a tiny whitespace nitpick, everything else looks perfect. Thanks!

@S11001001 Does it close #11159 as mentioned?

**Comment by S11001001** [comment]
> Does it close #11159 as mentioned?

@stefanobaghino-da Indeed it does.

**Pull request merged**